augur/tree.py

```diff
@@ -232,8 +232,23 @@ def random_string(n):
             tmp_line = line
             if line.startswith(">"):
                 num_seqs += 1
-                for c,v in escape_dict.items():
-                    tmp_line = tmp_line.replace(c,v)
+                pos = 1
+                for i in range(1, len(line)):
+                    l = line[i]
+                    #check if l is a character that IQ-tree changes
+                    # See IQtree function renameBool https://github.com/iqtree/iqtree2/blob/3bbc304263cb2f85574a9163e8f2e5c5b597a147/utils/tools.cpp#L585
+                    if l.isalnum()==False and l !='_' and l != '-' and l != '.' and l.isspace()== False:
```

Comment:
Hmm. Python's `str.isalnum()` isn't equivalent to the POSIX `isalnum()` used by IQ-TREE. It'd be best to match its behaviour since we're trying to cater to it here.

For `.isspace()`, it seems more sensible to me to avoid modifying anything but the id part (e.g. the part before the first whitespace) of the FASTA defline rather than skipping space mangling. I think IQ-TREE will only care about the id part.